### ISSUE TITLE
Ignore repositories which don't contain package.json

### ISSUE THREAD

**Issue by evilebottnawi** [singapore/renovate#376]
I learned the configuration (https://github.com/singapore/renovate/blob/master/docs/configuration.md), but could not find the option how to configure renovate to ignore repositories which don't contain `package.json`. In our self-hosted gitlab where is many repos, which don't contain `package.json`, because they written on other languages. Be great have this option.

**Comment by rarkins** [comment]
Are you seeing the same as this for current behaviour?
```
$ renovate --platform=gitlab --autodiscover
 INFO: Autodiscovering GitLab repositories
 INFO: Renovating repository (repository=renovate-bot/no-package-json)
 INFO: Processing package file (repository=renovate-bot/no-package-json, packageFile=package.json)
 WARN: No package.json content found - skipping (repository=renovate-bot/no-package-json, packageFile=package.json)
 INFO: Processing 0 dependency upgrade(s) (repository=renovate-bot/no-package-json)
 INFO: "Configure Renovate" PR needs to be closed first (repository=renovate-bot/no-package-json)
 INFO: Finished repository (repository=renovate-bot/no-package-json)
```

i.e. desired behaviour is that no "Configure Renovate" PR is created?

**Comment by rarkins** [comment]
Actually, the above was the second time I ran it. The first time was this:
```
$ renovate --platform=gitlab --autodiscover
 INFO: Autodiscovering GitLab repositories
 INFO: Renovating repository (repository=renovate-bot/no-package-json-2)
ERROR: Failed to process repository: Invalid reference name (400) (repository=renovate-bot/no-package-json-2)
 INFO: Finished repository (repository=renovate-bot/no-package-json-2)
```

**Comment by evilebottnawi** [comment]
@rarkins yep, at the moment I delete from many Merge Requests from repositories don't contain `package.json`.

**Comment by evilebottnawi** [comment]
@rarkins yep, first run (with `--autodiscover` or use specific repo - `company/repo`) create `Configure Renovate`
```
 INFO: Renovating repository (repository=company/repo)
 INFO: Processing package file (repository=company/repo, packageFile=package.json)
 WARN: No package.json content found - skipping (repository=company/repo, packageFile=package.json)
 INFO: Processing 0 dependency upgrade(s) (repository=company/repo)
 INFO: "Configure Renovate" PR needs to be closed first (repository=company/repo)
 INFO: Finished repository (repository=company/repo)
 INFO: Renovate finished
```
But it is create `Configure Renovate` and be great have option to avoid this behavior

**Comment by rarkins** [comment]
Should be fixed in 8.47.0 .. but it won't automatically close any existing Merge Requests created previously.

**Comment by evilebottnawi** [comment]
@rarkins thanks!
